source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -465,6 +471,24 @@ TemporaryFileStream CopyStreamToFile(Guid id, long length, BinaryReader reader)
             }
             return new TemporaryFileStream(path, log);
         }
+        
+        async Task<TemporaryFileStream> CopyStreamToFileAsync(Guid id, long length, Stream stream, CancellationToken cancellationToken)
```

Comment:
The `stream` parameter looks to be the member variable. That could get confusing in the method itself. Maybe delete `stream`?